src/components/SearchAndFilter/SearchAndFilter.tsx

```diff
@@ -13,19 +13,20 @@ const reduceSelectedFilters = (state: object, filterUpdate:filterUpdate) => {
   }
 }
 
-const SearchAndFilter: React.FC<filterProps> = ({data, columns, filterTypes}) => {
-  const [selectedFilters, dispatch] = useReducer(reduceSelectedFilters, {search: ''})
-  const [filteredData, setFilteredData] = useState<Array<object>>([])
-  const [searchField, setSearchField] = useState<string>('')
+const SearchAndFilter: React.FC<filterProps> = ({allData, columns, filterTypes}) => {
+  const [selectedFilters, dispatch] = useReducer(reduceSelectedFilters, {})
+  const [matchingFilterIds, setMatchingFilterIds] = useState<Array<string>>([])
+  const [searchTextBox, setSearchTextBox] = useState<string>('')
+  const [matchingSearchIds, setMatchingSearchIds] = useState<Array<string>>([])
```

Comment:
I decided to store IDs that matched search queries and filters for two reasons. 

1. I didn't want three copies of the data-set, even if some copies were partial
1. I had some challenges working with one set of matching data depending on whether the user searched first, or filtered first. This allows them to happen in either order and be compared as the other changes. 

This will be problematic when re-used, however, because as of now **there is no guarantee or failsafe if other data will have an IDs.** One solution could be to allow the programmer to declare which field is unique to each piece of data.